source/kernel/src/registry/mod.rs

```diff
@@ -0,0 +1,440 @@
+use core::any::TypeId;
+
+use mnemos_alloc::{containers::HeapFixedVec, heap::HeapGuard};
+use postcard::experimental::max_size::MaxSize;
+use serde::{de::DeserializeOwned, Deserialize, Serialize};
+use uuid::{uuid, Uuid};
+
+use crate::comms::{
+    bbq,
+    kchannel::{KProducer, LeakedKProducer},
+    rosc::Sender,
+};
+
+/// A partial list of known UUIDs of driver services
+pub static KNOWN_UUIDS: &[Uuid] = &[
+    //
+    // Kernel UUIDs
+    //
+
+    // SerialMux
+    uuid!("54c983fa-736f-4223-b90d-c4360a308647"),
+    // Simple Serial Port
+    uuid!("f06aac01-2773-4266-8681-583ffe756554"),
+];
```

Comment:
is an array of UUIDs really that useful? it seems like we might want to define this as a module with a bunch of constants in it, as well as an array. maybe something like this:
```suggestion
/// A partial list of known UUIDs of driver services
pub mod known_uuids {
     /// Kernel UUIDs
     pub mod kernel {
         pub const SERIAL_MUX: Uuid = uuid!("54c983fa-736f-4223-b90d-c4360a308647");
         pub const SIMPLE_SERIAL_PORT: Uuid = uuid!("f06aac01-2773-4266-8681-583ffe756554");
     }
     
     // In case you need to iterate over every UUID
     pub static ALL: &[Uuid] = &[
         kernel::SERIAL_MUX,
         kernel::SIMPLE_SERIAL_PORT,
     ];
}
```